src/index.ts

```diff
@@ -71,133 +75,104 @@ FirebaseAdmin.initializeApp(gcpAppOptions);
 const firestore = FirebaseAdmin.firestore();
 
 // discord.js on initialization
-client.on('ready', async () => {
-  // refresh discord.js commands
-  for (const [guildId] of client.guilds.cache) {
-    if (!client.user) {
-      console.error(client.user);
-      continue;
+globalClient.on('ready', async (client) => {
+  try {
+    // refresh discord.js commands
+    for (const [guildId] of client.guilds.cache) {
+      await deployCommands(
+        client.user.id,
+        guildId,
+        DISCORD_API_TOKEN,
+        firestore
+      );
     }
 
-    await deployCommands(client.user.id, guildId, DISCORD_API_TOKEN, firestore);
-  }
-
-  // setup activity status cycle
-  setRandomActivity(client);
-  const activityCronJob = new CronJob(RANDOM_ACTIVITY_CRON, () => {
+    // setup activity status cycle
     setRandomActivity(client);
-  });
-  activityCronJob.start();
+    const activityCronJob = new CronJob(RANDOM_ACTIVITY_CRON, () => {
+      setRandomActivity(client);
+    });
+    activityCronJob.start();
+  } catch (e: unknown) {
+    console.log('Uncaught exception:');
+    console.error(e);
+  }
 });
 
 // listen for discord.js command
-client.on('interactionCreate', async (interaction) => {
+globalClient.on('interactionCreate', async (interaction) => {
   if (!interaction.isCommand() || !interaction.isChatInputCommand()) return;
 
   try {
     const command = useCommand(interaction.commandName, [firestore]);
     await command.execute(interaction);
   } catch (e: unknown) {
-    // TODO Completely redo how we handle errors
-    console.error(e);
-
-    let errorMessage = 'ERROR: ';
+    let message: string | undefined;
 
-    if (e instanceof Error) {
-      errorMessage += e.message;
-    } else {
-      errorMessage += e;
+    if (e instanceof RustyBotCommandError) {
+      message = e.message;
     }
 
-    if (interaction.replied) {
-      await interaction.reply({ content: errorMessage, ephemeral: true });
-    } else {
-      await interaction.followUp({ content: errorMessage, ephemeral: true });
+    if (message) {
+      if (interaction.replied) {
+        await interaction.followUp({ content: message, ephemeral: true });
+      } else {
+        await interaction.reply({ content: message, ephemeral: true });
+      }
+
+      return;
     }
```

Comment:
Do we still want to interact/reply if the error is unknown just to say something went wrong?

Reply:
Yeah that's a good idea